### PR TITLE
[docs] Update index.rst for surrogate gradient

### PR DESCRIPTION
**Pull request by Routhleck**
## Description
Update `index.rst` for surrogate gradient and Corret some mistakes in `saving_and_loading.ipynb`

### CODE REVIEW

**Comment by chaoming0625** [comment]
There is one missing note for `Custom saving and loading`:


For customizing the saving, users can use:


```python

class YourClass(bp.BrainPyObject):
   def __save_state__(self):
     ....


```


For customizing the loading, users can use:


```python

class YourClass(bp.BrainPyObject):
   def __load_state__(self, state_dict):
     ....


```

Great work by the way.

https://brainpy.readthedocs.io/en/latest/tutorial_toolbox/saving_and_loading.html

**Review by chaoming0625 — APPROVED** [review]

**Comment by Routhleck** [comment]
> There is one missing note for `Custom saving and loading`:
> 
> For customizing the saving, users can use:
> 
> ```python
> class YourClass(bp.BrainPyObject):
>    def __save_state__(self):
>      ....
> ```
> 
> For customizing the loading, users can use:
> 
> ```python
> class YourClass(bp.BrainPyObject):
>    def __load_state__(self, state_dict):
>      ....
> ```
> 
> Great work by the way.
> 
> https://brainpy.readthedocs.io/en/latest/tutorial_toolbox/saving_and_loading.html

Understood👌
